model/project_ref.go

```diff
@@ -1255,6 +1304,180 @@ func (projectRef *ProjectRef) Upsert() error {
 	return err
 }
 
+// DefaultSectionToRepo modifies a subset of the project ref to use the repo values instead.
+// This subset is based on the pages used in Spruce.
+func (p *ProjectRef) DefaultSectionToRepo(section ProjectRefSection, caller string) error {
+	before, err := GetProjectSettingsEvents(p)
+	if err != nil {
+		return errors.Wrap(err, "error getting before project settings event")
+	}
+
+	switch section {
```

Comment:
> and pass in that struct containing the changes you want to make

I'm not sure what you mean by this. Should we be passing in a struct with the relevant ProjectRef fields already unset and then just call update? Or a map of things we want to change? I'm not sure it works really to make this very general, since each section updates different or multiple collections. 
We could make each one it's own function. I liked the idea of keeping it all together for organizational purposes.